Update support handbook (#363)
meltano/handbook: src/_support/index.md

permalink: /support/
---

## Slack Support

> Support on Slack is primarily provided by the community, but we strive to answer all questions within 24 hours. The [#support](https://meltano.slack.com/archives/C029CUM7ATD) Slack channel is where Meltano team members can request support from the wider team. If you see a support request that needs an assist, please share the relevant Slack thread in this channel.

Reviewer: "we strive to answer all questions within 24 hours" makes sense as something to tell our users, which is one purpose of the handbook, but for internal purposes I think this needs to be more precise about the how and the who, so that there can be no confusion over who are responsible for keeping this promise to our community (as much as possible, responding in 24h already won't work over the weekends for example). 

I _think_ the process here is that you and Sven check the main support channels (that would be good to list explicitly) every morning, look for unanswered threads from yesterday, and either directly respond or post them into `#support` for other team members to pick up. 

As for `#support`, I think we need to make it explicit who are expected to monitor that for assists, so that the process doesn't break down there and we can explicitly communicate that to the people in question. One downside of using #support for assists is that it also gets HubSpot support messages (email, live chat) now, which engineers aren't expected to respond to, so they may get conditioned into ignoring an unread support channel because it's not always for them. It may be better to ask for assists in #internal-engineering instead where engs are more likely to find it organically, or to pipe HubSpot messages elsewhere.

I don't care so much about a specific process as making sure things don't fall through the cracks and we have the edge cases covered, so super open to suggestions!

Once a request has been handled please add a ✅ emoji to the thread to let the team know that it's covered.

## Email

> Emails to `support@meltano.com` forward to HubSpot so they can be triaged, assigned, and managed.

Reviewer: These should also be piped into #support like live chats are, but apparently that didn't happen to that email from S...N... at S... that we were discussing that earlier today... Still worth documenting that that's the intended process though. And that all of those are forwarded by email to both you and me (right?) and that one of us will handle them within 24h.

### Responsible Disclosure of Security Vulnerabilities

> Emails to `security@meltano.com` also forward to Zendesk, and are automatically assigned to the Security group, which includes all current team members.

Reviewer: ```suggestion
Emails to `security@meltano.com` also forward to HubSpot, and are automatically assigned the Security Issue category.
```

As documented in our [Responsible Disclosure Policy](https://meltano.com/docs/responsible-disclosure.html), we will acknowledge receipt of a vulnerability report the next business day and strive to send the reporter regular updates about our progress.

## HubSpot Live Chat

The [Meltano.com website](https://www.meltano.com) is set up with live chat powered by HubSpot. Live chats are piped into #support on Slack and can be replied to within a thread. These are currently routed to `@Amanda Folson`. Pre-sales questions or RFPs are sent to `@Douwe Maan`.

## GitHub Issues/PRs

> We try to respond to issues and PRs created by the community within 24 hours. New PRs and issues across Meltano projects and the Singer ecosystem are posted to #singer-ecosystem-activity on Slack.

Reviewer: Activity on Singer tap/target repos goes into `#singer-ecosystem-activity`, but issues/PRs on the other Meltano projects (meltano/meltano, meltano/hub, meltano/sdk) does not. Those first two have their own channels (#meltano-repo-activity, #hub-repo-activity), but I don't think the SDK does, and those 2 channels are super noisy as they also include our team's activity, not just that by the community. Do we have a different process for those 3 repos we can document here? Or should we look into creating a new channel that only includes new community issues/PRs?

- Once a request has been handled please add a ✅ emoji to the thread to let the team know that it's covered.
- If the issue is related to our docs (or lackthereof) please open an issue or a PR to address the issue.

## Email

Emails to `support@meltano.com` forward to HubSpot so they can be triaged, assigned, and managed.

### Responsible Disclosure of Security Vulnerabilities

Emails to `security@meltano.com` also forward to HubSpot, and are automatically assigned the Security Issue category.
As documented in our [Responsible Disclosure Policy](https://meltano.com/docs/responsible-disclosure.html), we will acknowledge receipt of a vulnerability report the next business day and strive to send the reporter regular updates about our progress.

## HubSpot Live Chat

The [Meltano.com website](https://www.meltano.com) is set up with live chat powered by HubSpot. Live chats are piped into `#support` on Slack and can be replied to within a thread. These are currently routed to `@Amanda Folson`. Pre-sales questions or RFPs are sent to `@Douwe Maan`. We try to respond to these within 1 business day.

## GitHub Issues/PRs

> The DevRel team tries to respond to issues and PRs created by the community within 24 hours. New PRs and issues across Meltano projects are posted to `#meltano-repo-activity` and `#hub-repo-activity`, while Singer ecosystem are posted to `#singer-ecosystem-activity` on Slack. There is no SLA on Singer ecosystem issues and PRs, but we try to respond when there's something we think we can assist with.

Reviewer: I've created a new issue to create a `#sdk-repo-activity` channel as well: https://github.com/meltano/internal-general/issues/462, so we monitor all our official repos in Slack. All those channels are super noisy as they also include our team's activity, not just that by the community. Should we look into creating a new channel that only includes new community issues/PRs?

Also, would you mind documenting the process like you've done above? Sven indicated that he doesn't have bandwidth to monitor these, so if that works for you that'd be on you. I imagine it'd be a similar "every morning, respond to those from yesterday that didn't get a response from and eng or Taylor already, and ping the appropriate person".
